nixos/modules/services/hardware/display.nix

```diff
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.hardware.display;
+in
+{
+  ###### interface
+
+  options = {
+    hardware.display.edid.packages = lib.mkOption {
+      type = with lib.types; listOf package;
+      default = [ ];
+      description = lib.mdDoc ''
+        List of packages containing EDID binary files at `$out/lib/firmware/edid`.
+        Such files will be available for use in `drm.edid_firmware` kernel
+        parameter as `edid/<filename>`.
+      '';
+      example = lib.literalExpression ''
+        [
+          (pkgs.runCommand "edid-custom" {} '''
+            mkdir -p "$out/lib/firmware/edid"
+            base64 -d > "$out/lib/firmware/edid/custom1.bin" <<'EOF'
+            <insert your base64 encoded EDID file here `base64 < /sys/class/drm/card0-.../edid`>
+            EOF
+          ''')
+        ]
+      '';
+      apply = list:
+        if list == [ ] then null else
+        (pkgs.buildEnv {
+          name = "firmware-edid";
+          paths = list;
+          pathsToLink = [ "/lib/firmware/edid" ];
+          ignoreCollisions = true;
+        }) // {
+          compressFirmware = false;
+        };
+    };
+
+    hardware.display.edid.modelines = lib.mkOption {
+      type = with lib.types; attrsOf str;
+      default = { };
+      description = lib.mdDoc ''
+        Attribute set of XFree86 Modelines automatically converted
+        and exposed as `edid/<name>.bin` files in initrd.
+        See for more information:
+        - `hardware.display.edid.packages`
+        - https://en.wikipedia.org/wiki/XFree86_Modeline
+      '';
+      example = lib.literalExpression ''
+        {
+          "PG278Q_60" = "    241.50   2560 2608 2640 2720   1440 1443 1448 1481   -hsync +vsync";
```

Comment:
Those lines get rendered on search.nixos.org and I think for that they are bit long. Maybe removing some spaces would make them better to render.

Reply:
how about this?
```
"PG278Q_60" = builtins.concatStringsSep " " [
  "241.50"
  "2560 2608 2640 2720"
  "1440 1443 1448 1481"
  "-hsync +vsync"
];
```

Reply:
but to be honest this defeats the whole purpose of being visually similar to Modelines

Reply:
> Those lines get rendered on search.nixos.org and I think for that they are bit long.

@SuperSandro2000 I'd rather have properly formatted content which overflows than improperly formatted content.

@nazarewk That said, is there a good reason to have empty space at the beginning of each modeline? We could align them further to the left.

As an aside, would tabs be appropriate to separate the pseudo-columns?